hashes_and_arrays.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Library
+  @books = []
+
+  book1 = {
+    id: @books.length + 1,
+    year: 1956,
+    status: 'available',
+    author: 'Lucy Maud Montgomery',
+    name: 'Anne of Green Gables'
+  }
+  @books << book1
+
+  book2 = {
+    id: @books.length + 1,
+    year: 1985,
+    status: 'unavailable',
+    author: 'Edith Eva Eger',
+    name: 'The Choice'
+  }
+  @books << book2
+
+  book3 = {
+    id: @books.length + 1,
+    year: 2020,
+    status: 'available',
+    author: 'Edith Eva Eger',
+    name: 'The Gift'
+  }
+  @books << book3
+
+  def find_book(search_variable)
+    @books.any? { |book| book.values.include?(search_variable) }
+  end
+
+  def update_book_status(id, new_status)
+    book = @books.find { |book| book[:id] == id }
+    if book
+      book[:status] = new_status
+      puts "#{book[:name]} book new status is #{book[:status]}"
+    else
+      puts 'Book is not found in the Library'
+    end
+  end
+
+  def acquire_new_book(author, year, status, name)
+    book = {
+      id: @books.length + 1,
+      year: year,
+      status: status,
+      author: author,
+      name: name
+    }
+    @books << book
+  end
+
+  def delete_book(id)
+    @books.delete_if { |book| book[:id] == id }
+  end
+
+  def sort_books(sort_by, value = nil)
+    sorted_books = value ? @books.select { |book| book[sort_by.to_sym] == value } : @books
```

Comment:
Here we have two possibilities for sorting. 
1. Select by some value and then sort
2. Sort all books

The name of this method should be more descriptive. 
Maybe `sort_selected_books` or `select_sort_books`?  
or separate sorting from selecting

I don’t like my examples, maybe you can invent a better 😅
It is not a very necessary action here. 
Maybe just a tip for the future!

Reply:
Created separate default_book_sorting_by method for calling it inside sort_book method in case there is no value for selecting books